include/alpaka/core/Config.hpp

```diff
@@ -0,0 +1,230 @@
+/* Copyright 2023 Benjamin Worpitz, Matthias Werner, René Widera, Sergei Bastrakov, Jeffrey Kelling,
+ *                Bernhard Manfred Gruber, Jan Stephan
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#pragma once
+
+#ifdef __INTEL_COMPILER
+#    warning                                                                                                          \
+        "The Intel Classic compiler (icpc) is no longer supported. Please upgrade to the Intel LLVM compiler (ipcx)."
+#endif
+
+#define ALPAKA_VERSION_NUMBER(major, minor, patch)                                                                    \
+    ((((major) % 1000) * 100'000'000) + (((minor) % 1000) * 100000) + ((patch) % 100000))
```

Comment:
This will overflow if `major` is 22 or greater and is not explicitly marked as `unsigned int` or a `long`.
Would it make sense to force the promotion to `long` or `unsigned long` ?
_e.g._
```suggestion
    ((((major) % 1000ul) * 100'000'000ul) + (((minor) % 1000ul) * 100'000ul) + ((patch) % 100'000ul))```

Reply:
good catch, I will use you suggestion